src/duckling/entityeditor/entity-editor.component.ts

```diff
@@ -0,0 +1,74 @@
+import {
+    Component
+} from 'angular2/core';
+
+import {Entity, EntityKey, EntitySystemService, AttributeDefaultService, AttributeKey} from '../entitysystem';
+import {SelectionService, Selection} from '../selection';
+import {newMergeKey} from '../state';
+import {immutableAssign} from '../util';
+
+import {EntityComponent} from './entity.component';
+import {AttributeSelectorComponent} from './attribute-selector.component';
+
+/**
+ * Component that allows the user to modify an entity.
+ */
+@Component({
+    selector: "dk-entity-editor",
+    directives: [EntityComponent, AttributeSelectorComponent],
+    template: `
+        <div *ngIf="selection?.selectedEntity">
+            <span>
+                Entity {{selection.selectedEntity}}
+                <button md-button title="Delete Entity" (click)="deleteEntity()">
+                    Delete
+                </button>
+            </span>
+            <dk-entity-component *ngIf="selection?.selectedEntity"
+                [entity]="selection.entity"
+                (entityChanged)="onEntityChanged($event)">
+            </dk-entity-component>
+            <dk-attribute-selector *ngIf="selection?.selectedEntity"

```

Comment:
I changed dk-attribute-selector so that it displays a message when no new attributes are available.